### ISSUE TITLE
(vuejs) storybook documentation panel: code not understood

### ISSUE THREAD

**Issue by ksol** [storybookjs/storybook#13674]
Tested in storybook 6.1.10 with latest vue.js 2.x

Hello! Long story short, it seems whatever's handling the documentation aspect of the story (vue-docgen if I get it right) is "reading" the code, not interpreting it.

For instance, if the default value of a prop is a constant, it is the name of the constant that ends up in the storybook rather than the value. (cf below)

Another example, if I have a component that wraps another and re-uses some of its property, same thing, storybook does not understand what's happening. (cf below)

Finally, props, events and slots are put together in the same object (the argTypes if I followed right), so if I have a slot with the same name than a property (eg: `loading`), the prop won't be shown in the list.

This code :
```js
export default {
  name: "Button",
  props: {
    size: BaseButton.props.size,
    loading: BaseButton.props.loading,
    kind: {
      type: String,
      enum: BUTTON_KINDS,
      default: BUTTON_KIND_DEFAULT,
    },
}
```

ends up looking like this :

![Capture d’écran 2021-01-19 à 20 36 49](https://user-images.githubusercontent.com/228701/105086005-dfba7600-5a98-11eb-97e1-7beb4c0287d8.png)

Am I missing something somewhere, or is this known/expected?

Thanks for your time !

**Comment by shilman** [comment]
Yes, `vue-docgen-api` performs a static analysis. We just did a cleanup on the React side that initializes the default value with the interpreted code: https://github.com/storybookjs/storybook/pull/13937 

We should probably doe the same for Vue.

**Comment by vanessayuenn** [comment]
Hi! I am going through and cleaning up old issues in the repo. We just shipped a ton of improvements in Storybook 7.0, and I suspect this bug might be fixed already. Can you please try it out? If this issue persists with the latest version, please open a new bug report with a 7.x reproduction. Thank you so much! 🙏🏼
